src/source-nfq.c

```diff
@@ -887,17 +899,30 @@ int NFQParseAndRegisterQueues(const char *queues)
             return -1;
         }
 
-        for (uint16_t i = queue_start; i <= queue_end; i++) {
-            if (NFQRegisterQueue(i) != 0) {
-                return -1;
-            }
-        }
-    } else if (NFQRegisterQueue(queue_start) != 0) {
-        SCLogError(SC_ERR_INVALID_ARGUMENT, "queue(s) argument '%s' is not "
-                                        "valid", queues);
-        return -1;
+        num_queues = queue_end - queue_start + 1; // +1 due to inclusive range
     }
 
+    g_nfq_t = (NFQThreadVars *)SCCalloc(num_queues, sizeof(NFQThreadVars));
```

Comment:
We need to free allocated array at Suricata exit.

Reply:
Do you have an idea about a better place for that? ReceiveNFQThreadExitStats() seems to be the only option by now.

BTW, I've found another small leak - nfq_open() is called without nfq_close().